tests/pytests/test_multithread.py

```diff
@@ -78,198 +77,281 @@ def test_invalid_mt_config_combinations(env):
+            # the first iteration.
+            # At first iteration after reload we expect 2*n_vectors jobs to be executed as well (thread pool paused).
+            # At second iteration after reload we expect another n_vectors jobs to be executed (the second batch of
+            # vectors), after we resumed the workers, and then another 2*n_vector jobs upon loading. Overall 3*n_vectors
+            # jobs were added.
+            env.assertEqual(getWorkersThpoolStats(conn)['totalJobsDone'],
+                            (0 if it==0 else 2*n_vectors) if i==1 else (2*n_vectors if it==0 else 5*n_vectors),
+                            message=f"iteration {it+1} {'before' if i==1 else 'after'} loading")
+            assertInfoField(env, 'idx', 'num_docs', n_vectors*(it+1))
+            # Resume the workers thread pool, let the background indexing start (in the first iteration it is paused)
+            if i==1:
+                env.expect('FT.DEBUG', 'WORKER_THREADS', 'RESUME').ok()
             # At first, we expect to see background indexing, but after RDB load, we expect that all vectors
             # are indexed before RDB loading ends
-            # TODO: try making this not-flaky
-            if i == 2:
-                env.assertEqual(debug_info['BACKGROUND_INDEXING'], 0)
-            # env.assertEqual(debug_info['BACKGROUND_INDEXING'], 1 if i == 1 else 0)
+            debug_info = get_vecsim_debug_dict(env, 'idx', 'vector')
+            if i==2:
+                env.assertEqual(debug_info['BACKGROUND_INDEXING'], 0, message=f"iteration {it+1} after reloading")
+            env.expect('FT.DEBUG', 'WORKER_THREADS', 'drain').ok()
 
 
 def test_delete_index_while_indexing():
-    if CODE_COVERAGE:
-        raise unittest.SkipTest("Skipping since worker threads are not enabled")
-
     env = initEnv(moduleArgs='WORKER_THREADS 2 MT_MODE MT_MODE_FULL DEFAULT_DIALECT 2')
     conn = getConnectionByEnv(env)
     n_shards = env.shardsCount
-    n_vectors = 10000 * n_shards if not SANITIZER and not CODE_COVERAGE else 500 * n_shards
-    dim = 64
+    n_vectors = 100 * n_shards
+    dim = 4
     # Load random vectors into redis.
     env.expect('FT.CREATE', 'idx', 'SCHEMA', 'vector', 'VECTOR', 'HNSW', '8', 'TYPE', 'FLOAT32', 'M', '64',
                'DIM', dim, 'DISTANCE_METRIC', 'L2').ok()
+    env.expect(debug_cmd(), 'WORKER_THREADS', 'PAUSE').ok()
     load_vectors_to_redis(env, n_vectors, 0, dim)
+    assertInfoField(env, 'idx', 'num_docs', n_vectors)
+    env.expect(debug_cmd(), 'WORKER_THREADS', 'RESUME').ok()
+
     # Delete index while vectors are being indexed (to validate proper cleanup of background jobs in sanitizer).
-    debug_info = get_vecsim_debug_dict(env, 'idx', 'vector')
-    # TODO: try making this not-flaky
-    # env.assertEqual(debug_info['BACKGROUND_INDEXING'], 1)
     conn.execute_command('FT.DROPINDEX', 'idx')
```

Comment:
Do we have some statistics we can validate after the drop now?

Reply:
Added some checks